internal/autopilot-client/src/client.rs

```diff
@@ -0,0 +1,327 @@
+    api_key: SecretString,
+}
+
+impl AutopilotClient {
+    /// Creates a new builder for constructing an [`AutopilotClient`].
+    pub fn builder() -> AutopilotClientBuilder {
+        AutopilotClientBuilder::new()
+    }
+
+    /// Returns the base URL of the API.
+    pub fn base_url(&self) -> &Url {
+        &self.base_url
+    }
+
+    // -------------------------------------------------------------------------
+    // Health Endpoints
+    // -------------------------------------------------------------------------
+
+    /// Checks the API status.
+    ///
+    /// Returns `Ok(())` if the API is reachable.
+    pub async fn status(&self) -> Result<(), AutopilotError> {
+        let url = self.base_url.join("/status")?;
+        let response = self.http_client.get(url).send().await?;
+        self.check_response(response).await?;
+        Ok(())
+    }
+
+    /// Checks the API health.
+    ///
+    /// Returns detailed health information.
+    pub async fn health(&self) -> Result<serde_json::Value, AutopilotError> {
+        let url = self.base_url.join("/health")?;
+        let response = self.http_client.get(url).send().await?;
+        let response = self.check_response(response).await?;
+        let body = response.json().await?;
+        Ok(body)
+    }
+
+    // -------------------------------------------------------------------------
+    // Session Endpoints
+    // -------------------------------------------------------------------------
+
+    /// Lists sessions.
+    pub async fn list_sessions(
+        &self,
+        params: ListSessionsParams,
+    ) -> Result<ListSessionsResponse, AutopilotError> {
+        let url = self.base_url.join("/v1/sessions")?;
+        let response = self
+            .http_client
+            .get(url)
+            .headers(self.auth_headers()?)
+            .query(&params)
+            .send()
+            .await?;
+        let response = self.check_response(response).await?;
+        let body = response.json().await?;
+        Ok(body)
+    }
+
+    // -------------------------------------------------------------------------
+    // Event Endpoints
+    // -------------------------------------------------------------------------
+
+    /// Lists events for a session.
+    pub async fn list_events(
+        &self,
+        session_id: Uuid,
+        params: ListEventsParams,
+    ) -> Result<ListEventsResponse, AutopilotError> {
+        let url = self
+            .base_url
+            .join(&format!("/v1/sessions/{session_id}/events"))?;
+        let response = self
+            .http_client
+            .get(url)
+            .headers(self.auth_headers()?)
+            .query(&params)
+            .send()
+            .await?;
+        let response = self.check_response(response).await?;
+        let body = response.json().await?;
+        Ok(body)
+    }
+
+    /// Creates an event in a session.
+    ///
+    /// Use `Uuid::nil()` as the `session_id` to create a new session.
```

Comment:
(don't love this but internal and easy to change so noop for now)

Reply:
agree